File: 0001_0100/0054/0054.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, m = map(int, f.readline().split())
    maxmin = -1000
    d = [-1000] * m
    while n:
        n -= 1
        p = [int(x) for x in f.readline().split()]
        maxmin = max(maxmin, min(p))
        for i in range(m):
            if p[i] > d[i]:
                d[i] = p[i]
    q.write(' '.join((str(maxmin), str(min(d)))))

File: 0001_0100/0054/0054.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n, m, k int
	fmt.Fscan(f, &n, &m)
	maxmin, minmax := -1000, 1000
	p := make([]int, m)
	for i := 0; i < m; i++ {
		p[i] = -1000
	}
	for ; n > 0; n-- {
		min := 1000
		for i := 0; i < m; i++ {
			fmt.Fscan(f, &k)
			if k < min {
				min = k
			}
			if k > p[i] {
				p[i] = k
			}
		}
		if min > maxmin {
			maxmin = min
		}
	}
	for i := 0; i < m; i++ {
		if p[i] < minmax {
			minmax = p[i]
		}
	}
	fmt.Fprint(q, maxmin, " ", minmax)
}
